ir/expression.cpp

```diff
@@ -137,3 +137,45 @@ IR::Constant IR::Constant::operator-() const { return IR::Constant(-value); }
 IR::Constant IR::Constant::GetMask(unsigned width) {
     return (IR::Constant(1) << width) - IR::Constant(1);
 }
+
+const IR::Constant *IR::Constant::get(const IR::Type *t, big_int v, Util::SourceInfo si) {
```

Comment:
Please be careful we can only cache the values if the source info is empty. Otherwise we would need to take the source info into account. This kind of defeats the purpose of the cache. But we don't want to break error messages by the cache. Same applies below.

Reply:
I think only caching in case of an empty source info is a good approach. #4302 could solve this problem.